lib/src/http.dart

```diff
@@ -37,6 +44,20 @@ class _PubHttpClient extends http.BaseClient {
       : this._inner = inner == null ? new http.Client() : inner;
 
   Future<http.StreamedResponse> send(http.BaseRequest request) async {
+    if (_shouldAddMetadata(request)) {
+      request.headers["X-Pub-OS"] = Platform.operatingSystem;
+      request.headers["X-Pub-Command"] = PubCommandRunner.command.join(" ");
+      request.headers["X-Pub-Session-ID"] = _sessionId;
+
+      var environment = Platform.environment["PUB_ENVIRONMENT"];
+      if (environment != null) {
+        request.headers["X-Pub-Environment"] = environment;
```

Comment:
Generally speaking I think we should sanitize the data that goes into the headers (e.g. could this environment variable  have PII data in it?) and also what format we use: To avoid any issues we should e.g. restrict ourselves to ASCII-only header values.

If we want to have non-ASCII support we should base64 encode it.

Also the document kevin went out says something about one header with json value whereas this code seems to set different `X-Pub-*` headers. What's the status of this? 

@kevmoo @nex3